monai/apps/datasets.py

```diff
@@ -278,17 +280,18 @@ def get_properties(self, keys: Optional[Union[Sequence[str], str]] = None):
     def _generate_data_list(self, dataset_dir: str) -> List[Dict]:
```

Comment:
The `MedNISTDataset` is adapted from JoC team's MedNIST tutorial, it can set `val_frac` and `test_frac`, I think we should align with the original logic.
And as MedNIST contains many images, the existing random logic works fine for it.

Thanks.